src/persist-client/src/write.rs

```diff
@@ -178,7 +179,15 @@ where
     {
         trace!("WriteHandle::append lower={:?} upper={:?}", lower, upper);
         let batch = self.batch(updates, lower.clone(), upper.clone()).await?;
-        self.append_batch(batch, lower, upper).await
+        let res = self.append_batch(batch, lower, upper).await;
+        if !matches!(res, Ok(Ok(()))) {
+            // it's possible a client using `append_batch` continually fails if
+            // it's racing with another writer. that's perfectly fine, but we should
+            // be sure to heartbeat our writer explicitly if it's not getting a
+            // chance to renew its lease through `[Self::compare_and_append]`
+            self.maybe_heartbeat_writer().await;
```

Comment:
I think this needs to be `append_batch` instead of here. `append` is supposed to be simple sugar for calling `self.batch` + `self.append_batch`